graphics/src/CMakeLists.txt

```diff
@@ -27,6 +28,17 @@ gz_build_tests(
     gz-common${GZ_COMMON_VER}-testing
 )
 
+# Assimp doesn't offer preprocessor version, use cmake to set a compatibility
+# mode for versions below 5.2.0
+if(${GzAssimp_VERSION} STRLESS "5.2.0")
+  message("Warning, assimp below 5.2.0 detected, setting compatibility mode")
+  target_compile_definitions(${graphics_target} PRIVATE ASSIMP_COMPATIBILITY)
```

Comment:
The name `ASSIMP_COMPATIBILITY` isn't very informative. How about something like `GZ_ASSIMP_PRE_5_2_0`? That's following the pattern of [GZ_ZMQ_POST_4_3_1](https://github.com/gazebosim/gz-transport/blob/012a19dfb5ddbfe31411a909e18703b5a536b58a/include/gz/transport/Helpers.hh#L35).